gst/nnstreamer/tensor_query/tensor_query_common.c

```diff
@@ -521,19 +488,20 @@ query_tcp_receive (GSocket * socket, uint8_t * data, size_t size,
 {
   gsize bytes_received = 0;
   gssize rret;
-  GError *err;
-
+  GError *err = NULL;
   while (bytes_received < size) {
     rret = g_socket_receive (socket, (gchar *) data + bytes_received,
         size - bytes_received, cancellable, &err);
+    if (rret == 0) {
```

Comment:
In this case, Shouldn't we close the socket?

Reply:
Line 467 checks whether socket is closed and frees connection. It's not necessary for now

Reply:
Accept is understood. What if the client socket is closed during transferring data?

Reply:
I guess.. the closed socket would be still in connection queue, then it will be closed when accept function pops it from the queue.